### PR TITLE
[EC-449] Event log user for SCIM events

### PR DESCRIPTION
**Pull request by r-tome**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X] Other
```

## Objective

Currently in the Event Log, SCIM events are captured under the user “Unknown”. The user should be “SCIM”.
These changes parse property `SystemUser` from the events API response to display on the events table.

## Code changes

Related server changes on [this PR](https://github.com/bitwarden/server/pull/2306).

- **apps/web/src/app/organizations/manage/events.component.ts:** Displaying the `SystemUser` value on each Event row
- **libs/common/src/enums/eventSystemUser.ts:** Added new Enum to identity the possible `EventSystemUser` types
- **libs/common/src/models/response/eventResponse.ts:** Parsing the `SystemUser` property value from the API response

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/app/organizations/manage/events.component.ts [hunk 1]
```diff
@@ -111,6 +112,12 @@ export class EventsComponent extends BaseEventsComponent implements OnInit {
 
   protected getUserName(r: EventResponse, userId: string) {
     if (userId == null) {
+      if (r.systemUser != null) {
+        return {
+          name: EventSystemUser[r.systemUser],
+        };
+      }
+
```
Comment (MGibson1) [inline apps/web/src/app/organizations/manage/events.component.ts:120]: This only solved the userName in web viewer, please also update CLI and exporters (check out InstallationId on `EventResponse` and update both accordingly -- probably that model needs a `getUserName` helper)

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/organizations/manage/events.component.ts (hunk 1 above)
Comment (r-tome) [reply apps/web/src/app/organizations/manage/events.component.ts:120]: As we've discussed: the CLI does not have an Events command. The exporter has been updated.

File: libs/common/src/enums/event-system-user.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export enum EventSystemUser {
```
Comment (eliykat) [inline libs/common/src/enums/event-system-user.ts:1]: I think it's convenient to use the enum key in the UI, but it's unusual. A developer would not generally expect this (usually it's just for our convenience when reading the code), so I think we should flag it explicitly.

```suggestion
// Note: the enum key is used to describe the EventSystemUser in the UI. Be careful about changing it.
export enum EventSystemUser {
```

File: bitwarden_license/bit-web/src/app/providers/manage/events.component.ts [hunk 3]
```diff
@@ -82,8 +83,22 @@ export class EventsComponent extends BaseEventsComponent implements OnInit {
   }
 
   protected getUserName(r: EventResponse, userId: string) {
-    return userId != null && this.providerUsersUserIdMap.has(userId)
```
Comment (eliykat) [inline bitwarden_license/bit-web/src/app/providers/manage/events.component.ts:85]: This is used for Provider events, e.g. adding or removing a provider user. SCIM only applies to Organization events. So I don't think we need any changes here, other than to bring in the `installationId` stuff that you removed from the base component.

File: apps/web/src/app/organizations/manage/events.component.ts [hunk 4]
```diff
@@ -110,18 +111,26 @@ export class EventsComponent extends BaseEventsComponent implements OnInit {
...
-      return null;
+    if (r.installationId != null) {
+      return `Installation: ${r.installationId}`;
     }
 
-    if (this.orgUsersUserIdMap.has(userId)) {
-      return this.orgUsersUserIdMap.get(userId);
-    }
+    if (userId == null) {
+      if (r.systemUser != null) {
```
Comment (eliykat) [inline apps/web/src/app/organizations/manage/events.component.ts:119]: Please refactor to avoid this nested `if`.

**Review by eliykat — CHANGES_REQUESTED** [review]

**Review by eliykat — APPROVED** [review]

**Pull request merged**